src/main/java/com/spotinst/sdkjava/model/SpotAccountAdminService.java

```diff
@@ -4,13 +4,53 @@
 import com.spotinst.sdkjava.client.response.BaseSpotinstService;
 import com.spotinst.sdkjava.client.rest.*;
 import com.spotinst.sdkjava.exception.SpotinstHttpException;
+import com.spotinst.sdkjava.model.api.admin.account.ApiAccountAdmin;
 import org.apache.http.HttpStatus;
 
 import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
 import java.util.Map;
 
+
 public class SpotAccountAdminService extends BaseSpotinstService {
 
+    public static List<ApiAccountAdmin> listAccounts(String cloudAccountId, String authToken) throws SpotinstHttpException {
+
+        // Init retVal
+        List<ApiAccountAdmin> retVal = new LinkedList<>();
+
+        // Get endpoint
+        SpotinstHttpConfig config = SpotinstHttpContext.getInstance().getConfiguration();
+        String apiEndpoint = config.getEndpoint();
+
+        Map<String, String> queryParams = new HashMap<>();
+
+        // Add account Id Query param
+        if (cloudAccountId != null) {
+            queryParams.put("cloudAccountId", cloudAccountId);
+        }
+
+        // Get the headers for AWS.
+        Map<String, String> headers = buildHeaders(authToken);
+
+        //Build URI
+        String uri = String.format("%s/setup/account", apiEndpoint);
+
+        // Send the request.
+        RestResponse response = RestClient.sendGet(uri, headers, queryParams);
+
+        // Handle the response.
+        AccountAdminApiResponse allAccountsResponse = getCastedResponse(response, AccountAdminApiResponse.class);
+
+        if (allAccountsResponse.getResponse().getCount() > 0) {
```

Comment:
Why do you need that check? , maybe you meant : allAccountsResponse.getResponse().getCount() != null 
in that case the result will stay with the initial value (empty list)?

Reply:
1. The code will work without this null check also.
2. Still, I believe, getCount() adds some value, coz' it verifies that the 'count' field in the API response is not zero.
3. '!=null' check may not be the right thing to do, since, the return type is integer.

Reply:
use IsResponseSucceeded()